This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <info@kleisauke.nl>
Date: Wed, 15 Aug 2018 14:00:00 +0200
Subject: [PATCH 1/1] Add 'FILE_SHARE_DELETE' to the file sharing mode

This corresponds to Linux's behavior, where files can be renamed
and removed even when the file handles are still open.

diff --git a/goo/gfile.cc b/goo/gfile.cc
index 1111111..2222222 100644
--- a/goo/gfile.cc
+++ b/goo/gfile.cc
@@ -663,7 +663,7 @@ Goffset GooFile::size() const {
 GooFile* GooFile::open(const GooString *fileName) {
   HANDLE handle = CreateFileA(fileName->getCString(),
                               GENERIC_READ,
-                              FILE_SHARE_READ | FILE_SHARE_WRITE,
+                              FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
                               nullptr,
                               OPEN_EXISTING,
                               FILE_ATTRIBUTE_NORMAL, nullptr);
@@ -674,7 +674,7 @@ GooFile* GooFile::open(const GooString *fileName) {
 GooFile* GooFile::open(const wchar_t *fileName) {
   HANDLE handle = CreateFileW(fileName,
                               GENERIC_READ,
-                              FILE_SHARE_READ | FILE_SHARE_WRITE,
+                              FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
                               nullptr,
                               OPEN_EXISTING,
                               FILE_ATTRIBUTE_NORMAL, nullptr);
